sdk/eventgrid/Azure.ResourceManager.EventGrid/tests/EventGridManagementTestBase.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Azure.Core;
+using Azure.Core.TestFramework;
+using Azure.ResourceManager.Resources;
+using Azure.ResourceManager.TestFramework;
+using NUnit.Framework;
+using System.Threading.Tasks;
+
+namespace Azure.ResourceManager.EventGrid.Tests
+{
+    public class EventGridManagementTestBase : ManagementRecordedTestBase<EventGridManagementTestEnvironment>
+    {
+        protected ArmClient Client { get; private set; }
+
+        protected EventGridManagementTestBase(bool isAsync, RecordedTestMode mode)
+        : base(isAsync, mode)
+        {
+        }
+
+        protected EventGridManagementTestBase(bool isAsync)
+            : base(isAsync)
+        {
+        }
+
+        [SetUp]
+        public void CreateCommonClient()
+        {
+            Client = GetArmClient();
+        }
+
+        protected async Task<ResourceGroupResource> CreateResourceGroup(SubscriptionResource subscription, string rgNamePrefix, AzureLocation location)
+        {
+            string rgName = Recording.GenerateAssetName(rgNamePrefix);
+            ResourceGroupData input = new ResourceGroupData(location);
+            var lro = await subscription.GetResourceGroups().CreateOrUpdateAsync(WaitUntil.Completed, rgName, input);
```

Comment:
There are no live tests in this PR, but in case they will get added (similar to other mgmt libraries), is deletion handled in the tests?

Reply:
We are still digging that the framework can handle all clean up